test/cmd/newapp.sh

```diff
@@ -126,6 +158,10 @@ os::cmd::expect_success 'oc delete template ruby-helloworld-sample'
 os::cmd::expect_success_and_text 'oc new-app mysql --name=db' 'db'
 os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world -l app=ruby'
 os::cmd::expect_success 'oc delete all -l app=ruby'
+# check for error when template JSON file has errors
+jsonfile="${TMPDIR:-/tmp}/foo.json"
+echo '{"test"}' > "${jsonfile}"
+os::cmd::expect_failure_and_text "oc new-app '${jsonfile}'" "error: unable to load template file \"${jsonfile}\": invalid character '}' after object key"
```

Comment:
Hrm - that would mean the template file resolver has to have top weight,
which means a local file will *always* take precedence over any image from
anywhere.  So having a file called php in the local directory would prevent
you from being able to create php things.  I agree with the statement, but
I think we'd have to only look at things with .yaml and .json extensions to
give them that weight (php.json I can buy being weighted above everything
else, php i can't).

On Tue, Feb 2, 2016 at 11:04 AM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In test/cmd/newapp.sh
> <https://github.com/openshift/origin/pull/6670#discussion_r51588896>:
>
> > @@ -126,6 +158,10 @@ os::cmd::expect_success 'oc delete template ruby-helloworld-sample'
> >  os::cmd::expect_success_and_text 'oc new-app mysql --name=db' 'db'
> >  os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world -l app=ruby'
> >  os::cmd::expect_success 'oc delete all -l app=ruby'
> > +# check for error when template JSON file has errors
> > +jsonfile="${TMPDIR:-/tmp}/foo.json"
> > +echo '{"test"}' > "${jsonfile}"
> > +os::cmd::expect_failure_and_text "oc new-app '${jsonfile}'" "error: unable to load template file \"${jsonfile}\": invalid character '}' after object key"
>
> If you have a file in the current directory named template.json and run oc
> new-app template.json, what's the likelihood you meant a Docker image
> called "template.json"?
>
> I think in that case new-app should just say the file has invalid JSON (+
> details from the parser), and print nothing else.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6670/files#r51588896>.
>